### PR TITLE
Added functionality in FileUpload to handle YouTube links

### PR DESCRIPTION
**Pull request by LucasNVarella**
Also, added functionality in the backend to preview videos from a collection. However, this functionality has yet to be tested.

Currently, the page expects the user to copy-paste a regular (not embed) link to the video.

### CODE REVIEW

File: src/FileUpload.js [hunk 1]
```diff
@@ -37,10 +37,12 @@ class FileUpload extends Component {
             fileDoc: undefined,
         };
     };
-    handleTextChange = event => {
+
+    handleTextChange = async event => {
```
Comment (matticoli) [inline src/FileUpload.js]: Why async?

File: src/FileUpload.js [hunk 2]
```diff
@@ -37,10 +37,12 @@ class FileUpload extends Component {
...
         let fileDoc = this.state.fileDoc;
-        fileDoc.caption = event.target.value;
-        this.setState({fileDoc: fileDoc});
+        if (this.props.fileType === 'Video' && event.target.id.indexOf('multiline') === -1) fileDoc.url = event.target.value;
```
Comment (matticoli) [inline src/FileUpload.js]: Brackets pls

File: src/FileUpload.js [hunk 3]
```diff
@@ -37,10 +37,12 @@ class FileUpload extends Component {
...
+        else fileDoc.caption = event.target.value;
+        await this.setState({fileDoc: fileDoc});
```
Comment (matticoli) [inline src/FileUpload.js]: Why is await needed here?

**Comment by LucasNVarella** [comment]
Videos do not yet show up in previews - working on a fix.

**Pull request merged**